src/vfvm_solver.jl

```diff
@@ -266,97 +266,105 @@ function eval_and_assemble(system::AbstractSystem{Tv, Ti},
-    src=zeros(Tv,nspecies)
-
+    src   = zeros(Tv,nspecies)
+    bsrc  = zeros(Tv,nspecies)
     # arrays holding storage terms for old solution
-    oldstor=zeros(Tv,nspecies)
-    res_react=zeros(Tv,nspecies)
-    jac_react=zeros(Tv,nspecies,nspecies)
-    oldbstor=zeros(Tv,nspecies)
-
-    cfg_r=ForwardDiff.JacobianConfig(reactionwrap, Y, UK)
-    cfg_s=ForwardDiff.JacobianConfig(storagewrap, Y, UK)
-    cfg_br=ForwardDiff.JacobianConfig(breactionwrap, Y, UK)
-    cfg_bs=ForwardDiff.JacobianConfig(bstoragewrap, Y, UK)
-    cfg_flx=ForwardDiff.JacobianConfig(fluxwrap, Y, UKL)
+    oldstor   = zeros(Tv, nspecies)
+    res_react = zeros(Tv, nspecies)
+    jac_react = zeros(Tv, nspecies, nspecies)
+    oldbstor  = zeros(Tv, nspecies)
+
+    cfg_r     = ForwardDiff.JacobianConfig(reactionwrap, Y, UK)
+    cfg_s     = ForwardDiff.JacobianConfig(storagewrap, Y, UK)
+    cfg_br    = ForwardDiff.JacobianConfig(breactionwrap, Y, UK)
+    cfg_bs    = ForwardDiff.JacobianConfig(bstoragewrap, Y, UK)
+    
+    cfg_bflx  = ForwardDiff.JacobianConfig(bfluxwrap, Y, UKL)
+    cfg_flx   = ForwardDiff.JacobianConfig(fluxwrap, Y, UKL)
 
     # Inverse of timestep
     # According to Julia documentation, 1/Inf=0 which
     # comes handy to write compact code here for the
     # case of stationary problems.
-    tstepinv=1.0/tstep 
+    tstepinv = 1.0/tstep 
 
     
-    boundary_factors::Array{Tv,2}=system.boundary_factors
-    boundary_values::Array{Tv,2}=system.boundary_values
+    boundary_factors::Array{Tv,2} = system.boundary_factors
+    boundary_values::Array{Tv,2}  = system.boundary_values
+    
+    geom    = grid[CellGeometries][1]
+    csys    = grid[CoordinateSystem]
+    coord   = grid[Coordinates]
+    bgeom   = grid[BFaceGeometries][1]
+    nbfaces = num_bfaces(grid)
+    ncells  = num_cells(grid)
+    
+    cellnodes::Array{Ti,2}   = grid[CellNodes]
+    cellregions::Vector{Ti}  = grid[CellRegions]
+    bfacenodes::Array{Ti,2}  = grid[BFaceNodes]
+    bfaceregions::Vector{Ti} = grid[BFaceRegions]
+    cellx::Array{Ti,2}       = grid[CellNodes]
+    edgenodes::Array{Ti,2}   = local_celledgenodes(geom)
     
-    geom=grid[CellGeometries][1]
-    csys=grid[CoordinateSystem]
-    coord=grid[Coordinates]
-    cellnodes::Array{Ti,2}=grid[CellNodes]
-    cellregions::Vector{Ti}=grid[CellRegions]
-    bgeom=grid[BFaceGeometries][1]
-    bfacenodes::Array{Ti,2}=grid[BFaceNodes]
-    bfaceregions::Vector{Ti}=grid[BFaceRegions]
-    nbfaces=num_bfaces(grid)
-    ncells=num_cells(grid)
-
-
-    cellx::Array{Ti,2}=grid[CellNodes]
-    edgenodes::Array{Ti,2}=local_celledgenodes(geom)
     has_celledges=false
+    
     if haskey(grid,CellEdges)
-        cellx=grid[CellEdges]
-        edgenodes=grid[EdgeNodes]
-        has_celledges=true
+        cellx         = grid[CellEdges]
+        edgenodes     = grid[EdgeNodes]
+        has_celledges = true
     end
 
-    nn::Int=num_nodes(geom)
-    ne::Int=num_edges(geom)
+    bedgenodes = grid[BEdgeNodes]
```

Comment:
We need a type annotation here. In fact I need to check with ExtendableSparse why this is necessary at all. May be @chmerdon has a similar problem.

```
bedgenodes::Array{Ti,2}=...
```

Reply:
Yes, I once encountered similar problems when accessing grid component, type annotations usually help in these situations.

Reply:
Same here, now it runs.